contracts/modules/v2/Assessment.sol

```diff
@@ -35,238 +27,288 @@ contract Assessment is IAssessment, MasterAwareV2 {
+  ///
+  /// @param user        The address of the staker for which the rewards are withdrawn
+  /// @param untilIndex  The index until which the rewards should be withdrawn. Used if a large
+  ///                    number of assessments accumulates and the function doesn't fir in one
+  ///                    block, thus requiring multiple batched transactions.
+  function withdrawRewards(address user, uint104 untilIndex) external override
+  returns (uint withdrawn, uint withdrawUntilIndex) {
+    uint104 rewardsWithdrawnUntilIndex = stakeOf[user].rewardsWithdrawnUntilIndex;
+    {
+      uint voteCount = votesOf[user].length;
+      withdrawUntilIndex = untilIndex > 0 ? untilIndex : voteCount;
+      require(rewardsWithdrawnUntilIndex < voteCount, "No withdrawable rewards");
+    }
+
+    Vote memory vote;
+    Assessment memory assessment;
+    for (uint i = rewardsWithdrawnUntilIndex; i < withdrawUntilIndex; i++) {
+      vote = votesOf[user][i];
+      assessment = assessments[vote.assessmentId];
+      if (assessment.poll.end + config.payoutCooldownDays * 1 days >= block.timestamp) {
+        // Poll is not final
+        withdrawUntilIndex = i;
+        break;
+      }
+
+      withdrawn += assessment.totalReward * vote.stakedAmount /
+        (assessment.poll.accepted + assessment.poll.denied);
+    }
+
+    // This is the index where the next withdrawReward call will start iterating from
+    stakeOf[user].rewardsWithdrawnUntilIndex = uint104(withdrawUntilIndex);
+    ITokenController(getInternalContractAddress(ID.TC)).mint(user, withdrawn);
   }
 
-  function redeemIncidentPayout(uint104 incidentId, uint32 coverId, uint depeggedTokens)
-  external override onlyMember {
-    AssessmentIncidentsLib.redeemIncidentPayout(
-      internalContracts,
-      incidents[incidentId],
-      coverId,
-      depeggedTokens
-    );
-  }
 
-  function redeemCoverForDeniedClaim(uint coverId, uint claimId)
-  external override {
-    AssessmentClaimsLib.redeemCoverForDeniedClaim(
-    config,
-    internalContracts,
-    claims,
-    claimants,
-    coverId,
-    claimId
-    );
+  function startAssessment(uint totalAssessmentReward, uint assessmentDeposit) external
+  override onlyInternal returns (uint) {
+    assessments.push(Assessment(
+      Poll(
+        0, // accepted
+        0, // denied
+        uint32(block.timestamp), // start
+        uint32(block.timestamp + config.minVotingPeriodDays * 1 days) // end
+      ),
+      uint128(totalAssessmentReward),
+      uint128(assessmentDeposit)
+    ));
+    return assessments.length - 1;
   }
 
-  // [todo] Check how many times poll is loaded from storage
-  function castVote(uint8 eventType, uint104 id, bool accepted) external override onlyMember {
-    AssessmentVoteLib.castVote(
-    config,
-    eventType,
-    id,
-    accepted,
-    stakeOf,
-    votesOf,
-    hasAlreadyVotedOn,
-    claims,
-    incidents
+  function castVote(uint assessmentId, bool isAccepted) external override {
```

Comment:
also something to be careful with if the member switches addresses - we must make sure the stake is "transferred" to the new address